src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -6038,7 +6038,10 @@ final boolean complete(LogonCommand logonCommand, TDSReader tdsReader) throws SQ
         int dataLen = 0;
 
         // Denali --> TDS 7.4, Katmai (10.0) & later 7.3B, Prelogin disconnects anything older
-        if (serverMajorVersion >= 11) {
+        // TDS version 8 if strict mode
+        if (serverMajorVersion >= 17 && encryptOption.compareToIgnoreCase(EncryptOption.Strict.toString()) == 0) {
```

Comment:
Don't limit this to a specific server version.

1. This will certainly be extended to Azure SQL databases in the future.
2. I suspect (I don't know for sure) TDS 8 might even be backported to previous SQL versions in the future if TLS 1.2 is no longer considered secure before those versions fall out of support.

I would probably rename VER_SQL2022 to VER_TDS80, too.